### DISCUSSION TITLE
PNG images are broken in read me file

### DISCUSSION THREAD

**Discussion by kalaiselvan369** [community/community#56416] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

In readme file, the link to the images are not previewed properly even though the name and path to directory is correct.

Please check out this repo, https://github.com/kalaiselvan369/moneyspentios

**Comment by hrishikesh-kocharekar-dev** [comment]
Use absolute paths: Instead of using relative paths, try using absolute paths for the images. Absolute paths specify the complete URL or file path to the image. ex -
![Image](https://example.com/path/to/image.png)
By using absolute paths, you can ensure that the images are correctly previewed regardless of the file's location or the rendering environment.

Check image file extensions: Verify that the image file extensions match the actual file formats. For example, if an image is a PNG file, make sure the file extension in the Markdown link is `.png`.

or maybe you can try 👇🏻
Test the image links locally: Clone the repository and test the image links locally to see if the images are displayed correctly. This can help identify if the issue is specific to the GitHub rendering or if there is an issue with the image files themselves.

Use an image hosting service: Consider hosting the images on an image hosting service such as Imgur or a cloud storage provider. Upload the images to the hosting service and use the provided image URLs in the README file. This can help ensure that the images are consistently accessible and previewed correctly.

Preview the README file: Use a Markdown editor or a tool that allows you to preview Markdown files to see how the images are rendered. This can help you identify any formatting issues or discrepancies in the README file.

**Comment by airtower-luna** [comment]
What does the readme look like for you? All 4 images load just fine for me. You could also use the dev tools of your browser to check what happens with the requests.

**Reply by kalaiselvan369** [reply]
Okay. Let me check.

**Comment by kalaiselvan369** [comment]
I just took a screenshot after @hrishikesh-kocharekar-dev replied. The example images of @hrishikesh-kocharekar-dev  is not loading for me. Please find the below screenshot. This issue is occurring in all public repositories whichever containing the images in readme files.

![image](https://github.com/community/community/assets/14306248/81f60094-e954-4db7-9309-51880e8959c4)

**Comment by airtower-luna** [comment, marked as answer]
That example image isn't loading for me either, seems broken.

But if it's *everywhere* and not just your README, considering that your profile says you're in India it might be your provider blocking `raw.githubusercontent.com`:

* #32889

If it's that, using another DNS server (e.g. [Quad9](https://quad9.net/service/service-addresses-and-features/)) might help.

**Reply by kalaiselvan369** [reply]
You are correct @airtower-luna It is a network provider issue. I switched to another network provider and the image loading works. Thank you @airtower-luna

**Comment by jacksongatino** [comment]
If you're experiencing broken PNG images in a readme file, there could be a few possible reasons for this issue. Here are some steps you can take to troubleshoot and fix the problem:

Verify the image file: Double-check that the PNG images you're using are not corrupted or damaged. You can try opening them in an image viewer or editor to confirm their integrity.

Check the file paths: Ensure that the file paths specified in the readme file are correct and match the locations of the PNG images. Verify that the file names and extensions are accurate as well.

Use relative paths: Instead of using absolute file paths, it's recommended to use relative paths when referencing images in a readme file. Relative paths are more flexible and work irrespective of the project's location. For example, instead of /images/image.png, use ./images/image.png or ../images/image.png if the image is located in a subdirectory.

Image hosting: If you're hosting the PNG images on a website or image hosting service, ensure that the hosting service is reliable and the images are accessible. Check if the images can be loaded directly in a web browser.

Regarding your mention of a **[Pinterest image downloader](https://pintdow.com/pinterest-image-downloader/)**, it's important to note that automated scraping or downloading of images from Pinterest without explicit permission from the content owner or Pinterest itself may violate their terms of service. It's recommended to respect the platform's policies and only download images for personal use or when granted permission by the original creator.

If you still encounter issues with broken PNG images or have further questions, please provide more specific details or share relevant code snippets, and I'll be happy to assist you further.

**Comment by jacksongatino** [comment]
If the PNG **[Pinterest images](https://pintdow.com/pinterest-image-downloader/)** are appearing broken in the readme file, it could be due to a few reasons. Firstly, ensure that the image file paths in the readme are correct and that the images are located in the specified directories. Secondly, check if the images were properly uploaded to the repository or server where the readme is hosted. If the images were deleted or moved, they won't display correctly. Additionally, verify that the image files themselves are not corrupted or incomplete. If none of these issues are the cause, it's possible that there could be a problem with the rendering or compatibility of the PNG images with the platform or software being used to view the readme file. In such cases, try opening the readme in a different browser or image viewer to see if that resolves the issue.
